langchain4j-core/src/main/java/dev/langchain4j/data/document/DocumentLoader.java

```diff
@@ -20,10 +17,10 @@ private DocumentLoader() {
      * @return The loaded document.
      * @throws BlankDocumentException when the parsed {@link Document} is blank/empty.
      */
-    public static Document load(DocumentSource source, DocumentParser parser) {
+    static Document load(DocumentSource source, DocumentParser parser) {
         try (InputStream inputStream = source.inputStream()) {
             Document document = parser.parse(inputStream);
-            source.metadata().asMap().forEach((key, value) -> document.metadata().add(key, value));
+            source.metadata().toMap().forEach((key, value) -> document.metadata().add(key, value));
```

Comment:
We need to be careful with this change, we can break something

Reply:
I can't run integration tests to verify that. I hope this should be fine as long as tests are green.

Reply:
Comment for myself: check https://github.com/langchain4j/langchain4j/issues/1496